testsuite/classlibrary/TestBuffer.sc

```diff
@@ -245,4 +245,24 @@ TestBuffer : UnitTest {
 		buffer.free;
 	}
 
+	test_getToFloatArray {
+		var server, data;
+		this.bootServer;
+		server = Server.default;
+		data = [1.0, 5.0, 3.0, [9.0, -9.0], 2.0];
+
+		{
+			Duty.ar(SampleDur.ir, 0, Dseq(data, inf))
+		}.asBuffer(0.2, server, { |buffer|
+
+			buffer.getToFloatArray(0, data.size * 2, action: { |array|
+				this.assertEquals(array.as(Array), data.flop.lace,
+					"getToFloatArray should get correct data"
+				)
+			})
+		})
```

Comment:
The assert can't happen in an async callback; otherwise a failure will not be properly reported.

Reply:
Ah yes - actually I'm a little unsure how this is expected to play out.
I tried this:

```supercollider
test_getToFloatArray {
		var server, data, condition, inTime = false;
		condition = Condition.new;
		this.bootServer;
		server = Server.default;
		data = [1.0, 5.0, 3.0, [9.0, -9.0], 2.0];

		{
			Duty.ar(SampleDur.ir, 0, Dseq(data, inf))
		}.asBuffer(0.2, server, { |buffer|

			buffer.getToFloatArray(0, data.size * 2, action: { |array|
				condition.unhang;
				this.assertEquals(array.as(Array), data.flop.lace,
					"getToFloatArray should get correct data"
				);
				inTime = true;
			})
		});

		condition.hang;
		fork {
			3.wait;
			condition.unhang;
			this.assert(inTime, "reply was in time, less than 3 seconds")
		};


	}
```

but I never get the second assert back.